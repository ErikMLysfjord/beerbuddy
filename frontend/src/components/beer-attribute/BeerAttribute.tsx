import styles from "./BeerAttribute.module.css";

export interface BeerAttributeProps {
  icon: string;
  altText: string;
  attribute: string;
  value: string | number | undefined;
}

/**
 * A component for displaying a beer attribute.
 * This is used to display more detailed information about a beer.
 * @param icon - The icon to display.
 * @param altText - The alt text for the icon.
 * @param attribute - The attribute heading.
 * @param value - The value of the attribute.
 * @returns - The beer attribute component.
 */
const BeerAttribute = ({
  attribute,
  value,
  icon,
  altText,
}: BeerAttributeProps) => {
  return (
    <div className={styles.attributeContainer} aria-label={attribute}>
      <div className={styles.attributeHeaderContainer}>
        <img height={"32px"} width={"20px"} src={icon} alt={altText} />
        <h3 className={styles.attributeHeading}>{attribute}</h3>
      </div>
      <p className={styles.attributeText}>{value}</p>
    </div>
  );
};
export default BeerAttribute;
