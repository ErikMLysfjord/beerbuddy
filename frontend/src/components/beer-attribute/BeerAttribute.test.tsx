import { describe, it, expect } from "vitest";
import { render } from "@testing-library/react";
import BeerAttribute from "./BeerAttribute";
import { axe } from "jest-axe";

const Template = () => (
  <BeerAttribute
    icon={""}
    altText={"Alternative text"}
    attribute={"Test"}
    value={5}
  />
);

describe("BeerAttribute", () => {
  it("is accessible", async () => {
    const { container } = render(<Template />);
    expect(await axe(container)).toHaveNoViolations();
  });

  it("renders correctly", () => {
    const { container } = render(<Template />);
    expect(container).toMatchSnapshot();
  });
});
