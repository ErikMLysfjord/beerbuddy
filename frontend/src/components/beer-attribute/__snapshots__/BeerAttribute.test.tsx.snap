// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BeerAttribute > renders correctly 1`] = `
<div>
  <div
    aria-label="Test"
    class="_attributeContainer_eeaf5b"
  >
    <div
      class="_attributeHeaderContainer_eeaf5b"
    >
      <img
        alt="Alternative text"
        height="32px"
        src=""
        width="20px"
      />
      <h3
        class="_attributeHeading_eeaf5b"
      >
        Test
      </h3>
    </div>
    <p
      class="_attributeText_eeaf5b"
    >
      5
    </p>
  </div>
</div>
`;
